Library/Homebrew/cmd/outdated.rb

```diff
@@ -62,13 +62,14 @@ def run
           end
 
           json = {
-            "formulae" => json_info(formulae),
-            "casks"    => json_info(casks),
+            formulae: json_info(formulae),
+            casks:    json_info(casks),
           }
-          puts JSON.pretty_generate(json)
+          # json v2.8.1 is inconsistent it how it renders empty arrays,
+          # so we use `[]` for consistency:
+          puts JSON.pretty_generate(json).gsub(/\[\n\n\s*\]/, "[]")
```

Comment:
Seems like a bug that should be reported upstream?

Reply:
I'm working on it. It's actually a bug that's been around a while, you can even see it in our fixtures: https://github.com/Homebrew/brew/blob/1d19230268f8de233b503177f0d4a5a4bb2072c7/Library/Homebrew/test/support/fixtures/cask/everything.json#L4-L6
There's also an SO question around the issue: https://stackoverflow.com/questions/58361230/avoiding-extra-lines-in-ruby-json-pretty-generate

The unusual thing (to me) is seeing different output on substantially similar inupts.